mmv1/third_party/terraform/utils/header_transport.go

```diff
@@ -11,14 +11,16 @@ type headerTransportLayer struct {
 	baseTransit http.RoundTripper
 }
 
-func newTransportWithHeaders(baseTransit http.RoundTripper) headerTransportLayer {
+func newTransportWithHeaders(baseTransit http.RoundTripper, configRequestReason string) headerTransportLayer {
 	if baseTransit == nil {
 		baseTransit = http.DefaultTransport
 	}
 
 	headers := make(http.Header)
 	if requestReason := os.Getenv("CLOUDSDK_CORE_REQUEST_REASON"); requestReason != "" {
```

Comment:
Once we're using a defaultfunc, we can remove the env variable logic here and always draw from config. Also, we'd prefer the opposite precedence- config over env var (which the defaultfunc will do)